source/patterns/@aws-solutions-constructs/core/lib/glue-job-helper.ts

```diff
@@ -168,19 +182,22 @@ export function deployGlueJob(scope: Construct, glueJobProps: glue.CfnJobProps,
   };
 
   const _newGlueJobProps: glue.CfnJobProps = overrideProps(defaults.DefaultGlueJobProps(_jobRole!, glueJobProps,
-    _glueSecurityConfigName, _jobArgumentsList), glueJobProps);
-
-  let _scriptLocation: string;
-  if (isJobCommandProperty(_newGlueJobProps.command)) {
-    if (_newGlueJobProps.command.scriptLocation) {
-      _scriptLocation = _newGlueJobProps.command.scriptLocation;
-    } else {
-      throw Error('Script location has to be provided as an s3 Url location. Script location cannot be empty');
+    _glueSecurityConfigName, _jobArgumentsList, etlCodeAsset), glueJobProps);
+  if (etlCodeAsset) {
+    etlCodeAsset.grantRead(_jobRole);
+  } else {
+    let _scriptLocation: string;
+    if (isJobCommandProperty(_newGlueJobProps.command)) {
+      if (_newGlueJobProps.command.scriptLocation) {
+        _scriptLocation = _newGlueJobProps.command.scriptLocation;
+      } else {
+        throw Error('Script location has to be provided as an s3 Url location. Script location cannot be empty');
```

Comment:
We're throwing an error and descriptive message around this in multiple places (here and line 251). Let's check our inputs at the construct layer:

Check that only 1 of script location and asset are specified - if neither or both throw an error.
if script location is specified, do a regex check for validity and throw if it fails.

Then here just throw an error in getS3ArnFromS3Url if the value is invalid or empty with the message "Invalid S3 url format".